### PR TITLE
Refactoring docs

### LINKED ISSUE #2203: Refactoring architecture doc
See #2177

### PR DESCRIPTION
**Pull request by gterzian**
Part of #2177

Close #2203

### CODE REVIEW

**Comment by gterzian** [comment]
Life of a block(local production), looks like:

[](https://mermaid-js.github.io/mermaid-live-editor/edit#eyJjb2RlIjoiICBqb3VybmV5XG4gICAgdGl0bGUgTGlmZSBvZiBhIGJsb2NrKGxvY2FsIHByb2R1Y3Rpb24pXG4gICAgc2VjdGlvbiBGYWN0b3J5XG4gICAgICBQcm9kdWNlIGJsb2NrOiA3OiBDb25zZW5zdXMsIFN0b3JhZ2UsIFBPU1xuICAgICAgTm90aWZ5IG5ldyBibG9jazogNzogQ29uc2Vuc3VzXG4gICAgc2VjdGlvbiBDb25zZW5zdXNcbiAgICAgIFByb2Nlc3MgbmV3IGJsb2NrOiA3OiBTdG9yYWdlXG4gICAgICBOb3RpZnkgbmV3IGdyYXBoOiA3OiBFeGVjdXRpb24sIE5ldHdvcmtcbiAgICBzZWN0aW9uIEV4ZWN1dGlvblxuICAgICAgRXhlY3V0ZSBncmFwaDogNzogU3RvcmFnZVxuICAgICAgVXBkYXRlIGxlZGdlcjogNzogU3RvcmFnZSwgUE9TXG4gICAgICAiLCJtZXJtYWlkIjoie1xuICBcInRoZW1lXCI6IFwiZGVmYXVsdFwiXG59IiwidXBkYXRlRWRpdG9yIjpmYWxzZSwiYXV0b1N5bmMiOnRydWUsInVwZGF0ZURpYWdyYW0iOmZhbHNlfQ)

**Comment by gterzian** [comment]
[](https://mermaid-js.github.io/mermaid-live-editor/edit/#eyJjb2RlIjoiICBqb3VybmV5XG4gICAgdGl0bGUgTGlmZSBvZiBhIGJsb2NrKGluY29taW5nIGZyb20gbmV0d29yaylcbiAgICBzZWN0aW9uIE5ldHdvcmtcbiAgICAgIFByb2Nlc3MgYmxvY2s6IDc6IFN0b3JhZ2UsIFBPU1xuICAgICAgTm90aWZ5IG5ldyBibG9jazogNzogQ29uc2Vuc3VzXG4gICAgc2VjdGlvbiBDb25zZW5zdXNcbiAgICAgIFByb2Nlc3MgbmV3IGJsb2NrOiA3OiBTdG9yYWdlXG4gICAgICBOb3RpZnkgbmV3IGdyYXBoOiA3OiBFeGVjdXRpb24sIE5ldHdvcmtcbiAgICBzZWN0aW9uIEV4ZWN1dGlvblxuICAgICAgRXhlY3V0ZSBncmFwaDogNzogU3RvcmFnZVxuICAgICAgVXBkYXRlIGxlZGdlcjogNzogU3RvcmFnZSwgUE9TXG4gICAgICAiLCJtZXJtYWlkIjoie1xuICBcInRoZW1lXCI6IFwiZGVmYXVsdFwiXG59IiwidXBkYXRlRWRpdG9yIjpmYWxzZSwiYXV0b1N5bmMiOnRydWUsInVwZGF0ZURpYWdyYW0iOmZhbHNlfQ)

File: massa-consensus/architecture.md [hunk 1]
```diff
@@ -62,4 +62,18 @@
...
+
+### Various
+1. Best parents(writer: Consensus)
+2. Draws(writer: POS)
+3. Ledger(writer: Executor)
+4. Final blocks(writer: Executor)
+
+### Storage
+1. Blocks(writer: Network, Production)
+2. Endorsements(writer: Network)
```
Comment (damip) [inline massa-consensus/architecture.md:76]: And factory (production)

File: massa-consensus/architecture.md [hunk 2]
```diff
@@ -62,4 +62,18 @@
...
+3. Operations(writer: Network)
```
Comment (damip) [inline massa-consensus/architecture.md:77]: Api

**Comment by gterzian** [comment]
@AureliaDolo @damip @adrien-zinger @yvan-sraka 

I have started writing more about "how components run", based on shared data and notification on that shared data. To be honest it may turn out to be a bit more complicated(for example network, which should be notified both by consensus when blocks need to be propagated, but also when new stuff comes in over the network), and so we may have to add components and/or shared data to reflect this. 

Work in progress and you may take a look and comment whenever you want...

**Comment by yvan-sraka** [comment]
That's neat ✨! Not sure to get what smileys 😊 stands for?

File: massa-consensus/architecture.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+### Slot
+
+Dedicated thread running a timer and writint the `Slot`, nofitying Factory on the condvar.
+
+### Factory
+Waits on a slot timer(and shutdown).
+
+1. Wake-up at each slot
+2. Read draws.
+3. If drawn: read best parents and produce.
```
Comment (AureliaDolo) [inline massa-consensus/architecture.md:157]: it may produce a block and/or an endorsement right ?

Comment (gterzian) [reply massa-consensus/architecture.md:157]: Yes I think you're right, I'll update, thanks.

**Comment by gterzian** [comment]
@massalabs/core-team So I'm starting to think that we do really need various "message queues" between components, which have to be separate from shared data like the proposed `Storage`(because different component want to be notified to different things related to it). 

Whether we do that with condvars, or channels, it's I think the same, and we might as well focus on introducing some shared data to avoid cloning, like with `Storage`, and also split or consolidate components, like with `Factory`, but leave the basic structure using channels. We can still switch to threads and threaded channels where necessary off-course.

**Pull request closed without merging**